### PR TITLE
Move the GlossaryTerm model

### PR DESCRIPTION
**Pull request by willbarton**
This is an attempt to resolve a circular dependency we have in our migrations for `ask_cfpb` and `v1`. 

The `GlossaryTerm` model is closely tied to Ask CFPB per @higs4281, so I've moved it to the `ask_cfpb` app. 

This is accomplished with a migration that alters the table name in the database and then "deletes" the `v1.GlossaryTerm` model for the purposes of the Django migration state. Another migration then adds the `ask_cfpb.GlossaryTerm` model to the Django migration state without performing any database actions. This preserves both the existing data and the default table naming for models in Django.

I've left the `v1.wagtail_hooks` alone, which adds `GlossaryTerm` to the snippets menu. The `GlossaryTermAdmin` is still defined there. That seems to be the right decision for causing the least amount of user-facing disruption or adding any additional complexity to snippets for now. But it does simply move the problem of code organization up a level above models.

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [ ] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### CODE REVIEW

**Comment by higs4281** [comment]
Sorry I didn't notice that migration state was not happy.

**Comment by higs4281** [comment]
I think adding a no-op table alteration to 0035 does it.

```python
    # ...
        migrations.AlterUniqueTogether(
            name='glossaryterm',
            unique_together=set([('portal_topic', 'name_en')]),
        ),
        migrations.AlterModelTable(
            name='glossaryterm',
            table=None,
        ),
    ]

```

**Comment by willbarton** [comment]
@higs4281 I fixed the Travis migration warning — it turns out if you specify the table in the migration but not in the model (because it's the default table), it thinks you've removed the custom table name.

File: cfgov/ask_cfpb/migrations/0035_move_glossaryterm.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            name='glossaryterm',
+            unique_together=set([('portal_topic', 'name_en')]),
+        ),
+    ]
+
+    # GlossaryTerm is moving from v1.models.snippets to
+    # ask_cfpb.models.snippets. Because of this, this migration will only make
+    # changes to the migration state, and not to the database itself.
+    operations = [
+        migrations.SeparateDatabaseAndState(state_operations=state_operations)
```
Comment (schbetsy) [inline cfgov/ask_cfpb/migrations/0035_move_glossaryterm.py]: I tried looking up what `SeparateDatabaseAndState` does, and found this 🤷‍♀ :
Screen Shot 2019-12-13 at 10 23 14 AM

**Comment by cwdavies** [comment]
I verified using:
```
$ ./refresh-data.sh production_django.sql.gz
Validating local dump file
Importing refresh db
Running any necessary migrations
Operations to perform:
  Apply all migrations: admin, agreements, ask_cfpb, auth, comparisontool, contenttypes, countylimits, data_research, flags, hmda, jobmanager, paying_for_college, prepaid_agreements, ratechecker, regulations3k, retirement_api, sessions, taggit, teachers_digital_platform, v1, wagtailadmin, wagtailcore, wagtaildocs, wagtailembeds, wagtailforms, wagtailimages, wagtailinventory, wagtailredirects, wagtailsharing, wagtailusers
Running migrations:
  Applying v1.0194_english_homepage_cleanup... OK
  Applying v1.0195_move_glossaryterm... OK
  Applying ask_cfpb.0035_move_glossaryterm... OK
  Applying ask_cfpb.0035_auto_20191206_1118... OK
  Applying ask_cfpb.0036_merge_20191213_1009... OK
  Applying jobmanager.0019_auto_20191206_1118... OK
  Applying paying_for_college.0006_auto_20191206_1118... OK
  Applying regulations3k.0029_auto_20191206_1118... OK
```

**Comment by schbetsy** [comment]
Does this solve the issue of the circular dependencies that were preventing us from re-creating the migrations?

File: cfgov/ask_cfpb/migrations/0035_move_glossaryterm.py (hunk 1 above)
Comment (willbarton) [reply cfgov/ask_cfpb/migrations/0035_move_glossaryterm.py]: Heh. I *think* I'm sure 😛

**Comment by willbarton** [comment]
@schbetsy Yes, it does.

**Comment by schbetsy** [comment]
Then I'm okay with merging. 👍

**Pull request merged**